make_wheels.py

```diff
@@ -129,50 +131,65 @@ def fetch_zig_version_info():
-        if zig_archive_hash != expected_hash:
-            print(zig_download, "SHA256 hash mismatch!")
-            raise AssertionError
-        print(f'{hashlib.sha256(zig_archive).hexdigest()} {zig_url}')
-
-    wheel_path = write_ziglang_wheel('dist/',
-        version=effective_zig_version.replace('-', '.') + python_version_suffix,
-        platform=python_platform,
-        archive=zig_archive)
-    with open(wheel_path, 'rb') as wheel:
-        print(f'  {hashlib.sha256(wheel.read()).hexdigest()} {wheel_path}')
+
+def fetch_and_write_ziglang_wheels(
+    outdir='dist/', zig_version='master', python_version_suffix='', platforms=tuple()
+):
+    Path(outdir).mkdir(exist_ok=True)
+    if not platforms:
+        platforms = list(ZIG_PYTHON_PLATFORMS)
+    zig_versions_info = fetch_zig_version_info()
+
+    if zig_version == 'latest':
+        zig_version = [version for version in zig_versions_info if version != 'master'][0]
+
+    try:
+        zig_version_info = zig_versions_info[zig_version]
+    except KeyError:
+        print(f"Invalid version, valid values: {list(zig_versions_info)}")
+        raise
+
+    effective_zig_version = zig_version_info.get('version', zig_version)
+
+    for zig_platform in platforms:
+        python_platform = ZIG_PYTHON_PLATFORMS[zig_platform]
+        if zig_platform not in zig_version_info:
+            print(f"{zig_platform} not present for "
+                  f"version {zig_version} / {effective_zig_version}")
+            continue
+        zig_download = zig_version_info[zig_platform]
+        zig_url = zig_download['tarball']
+        expected_hash = zig_download['shasum']
+
+        with urllib.request.urlopen(zig_url) as request:
+            zig_archive = request.read()
+            zig_archive_hash = hashlib.sha256(zig_archive).hexdigest()
+            if zig_archive_hash != expected_hash:
+                print(zig_download, "SHA256 hash mismatch!")
+                raise AssertionError
+            print(f'{hashlib.sha256(zig_archive).hexdigest()} {zig_url}')
+
+        wheel_path = write_ziglang_wheel(outdir,
+            version=effective_zig_version.replace('-', '.') + python_version_suffix,
+            platform=python_platform,
+            archive=zig_archive)
+        with open(wheel_path, 'rb') as wheel:
+            print(f'  {hashlib.sha256(wheel.read()).hexdigest()} {wheel_path}')
+
+def get_argparser():
+    parser = argparse.ArgumentParser(prog=__file__, description="Repackage official Zig downloads as Python wheels")
+    parser.add_argument('--version', default='latest',
+                        help="version to package, use `latest` for latest release, `master` for nightly build")
+    parser.add_argument('--suffix', default='', help="wheel version suffix")
+    parser.add_argument('--outdir', default='dist/', help="target directory")
+    parser.add_argument('--platform', action='append', choices=list(ZIG_PYTHON_PLATFORMS.keys()), default=[],
```

Comment:
I'd like the default to stay the same as before.

Reply:
Sorry, I actually mean building all of the platforms if none are specified.

Reply:
Added.
I guess with a bit of Github Action automation, packaging and pushing new versions to PyPI could be automated.

Reply:
Ah sorry, it should do that already (or before the second commit). Line 139, if no platforms are specified, it iters thru all.

Reply:
Ah okay, I misread then. I think defaulting to latest is fine then.

For GitHub automation, take a look at the YoWASP project which does exactly that.

Reply:
Okay, thanks. I'll take a look.

I've removed the last commit again.